fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/properties/LuceneRecordContextProperties.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * LuceneRecordContextProperties.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2021 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb.properties;
+
+public final class LuceneRecordContextProperties {
+    /**
+     * Get a builder for a boolean property key to control whether the compression of lucene index is enabled.
+     *
+     * @return the builder to populate a property key with value supplier, or as a key to find the property from {@link FDBRecordContextProperties#getProperty(FDBPropertyKey.Builder)}
+     */
+    public static FDBPropertyKey.Builder<Boolean> getLuceneIndexCompressionEnabled() {
+        return FDBPropertyKey.booleanPropertyBuilder("com.apple.foundationdb.record.provider.foundationdb.lucene.compressionEnabled", Boolean.FALSE);
```

Comment:
As implemented, this will create a new `FDBPropertyKey.Builder` instance every single time it is called. However, at the moment, it's unsafe to store the `FDBPropertyKey.Builder` as a static member because that is a mutable class. If we could store the `FDBPropertyKey` (i.e., not the builder), then I think we would be able to save the copies

Reply:
Fixed.